src/parse/matches/arg_matches.rs

```diff
@@ -129,9 +130,11 @@ impl ArgMatches {
     }
 
     /// Gets the lossy value of a specific argument. If the argument wasn't present at runtime
-    /// it returns `None`. A lossy value is one which contains invalid UTF-8 code points, those
+    /// it returns `None`. A lossy value is one which contains invalid UTF-8, those
     /// invalid points will be replaced with `\u{FFFD}`
     ///
+    /// *NOTE:* Requires having set [`ArgSettings::AllowInvalidUtf8`][crate::ArgSettings::AllowInvalidUtf8].
```

Comment:
This doesn't need it to be set, right? `value_of_lossy` still works with full utf8 values. Similarily for below places where you added the same note.

Reply:
Yes, you *can* use `value_of_lossy` and `value_of_os` without `AllowInvalidUtf8` but that is automatically a code smell.   The access code is communicating to developers that there might be invalid UTF-8 but the arg definition says there won't.  To keep the documentation simple, I focused on the likely intended use case.

Reply:
What about using the word "recommend" instead of "require" then?

Reply:
Personally, I think `required` is more realistic to what is happening rather than leaving it open for people having to figure out if the advice applies to them.  However, not important enough, I'll change it.